autoload/phpactor/input.vim

```diff
@@ -32,19 +33,12 @@ function! phpactor#input#choice(label, choices, ResultHandler)
 
     for choiceLabel in keys(a:choices)
         let buffer = []
-        let foundShortcut = v:false
 
-        for char in split(choiceLabel, '\zs')
-            if v:false == foundShortcut && -1 == index(usedShortcuts, tolower(char))
-                call add(buffer, '&')
-                let foundShortcut = v:true
-                call add(usedShortcuts, tolower(char))
-            endif
-
-            call add(buffer, char)
-        endfor
-
-        let confirmLabel = join(buffer, "")
+        if !empty(a:keyMap) && has_key(a:keyMap, choiceLabel) && !empty(a:keyMap[choiceLabel])
+            let confirmLabel = s:determineConfirmLabelFromPreference(choiceLabel, a:keyMap[choiceLabel])
+        else
+            let confirmLabel = s:determineConfirmLabel(choiceLabel)
+        endif
 
         call add(list, confirmLabel)
         call add(choices, choiceLabel)
```

Comment:
`choices` consist of all keys of `a:choises` and nothing else. It seems than you can simply assign `choices` to `keys(a:choices)` before this loop and iterate through it, isn't it?

Reply:
indeed, well spotted.